code/modules/reagents/reagent_containers/pill.dm

```diff
@@ -47,13 +47,16 @@
 	return on_consumption(M, user)
 
 ///Runs the consumption code, can be overriden for special effects
-/obj/item/reagent_containers/pill/proc/on_consumption(mob/M, mob/user)
-	if(icon_state == "pill4" && prob(5)) //you take the red pill - you stay in Wonderland, and I show you how deep the rabbit hole goes
-		addtimer(CALLBACK(GLOBAL_PROC, GLOBAL_PROC_REF(to_chat), M, span_notice("[pick(strings(REDPILL_FILE, "redpill_questions"))]")), 50)
+/obj/item/reagent_containers/pill/proc/on_consumption(mob/consumer, mob/giver)
+	if(icon_state == "pill4") //you take the red pill - you stay in Wonderland, and I show you how deep the rabbit hole goes
+		if(consumer.GetComponent(/datum/component/avatar_connection))
+			SEND_SIGNAL(consumer, COMSIG_BITRUNNER_ALERT_SEVER) //if your bitrunning avatar somehow manages to acquire and consume a red pill, they will be ejected from the Matrix
```

Comment:
so that the else statement won't send the message to an empty husk

Reply:
not sure why you're doing this either - and please make it a totally unique signal for this interaction rather than the one used on clicking the exit alert

Reply:
sure, I'll do that tomorrow

Reply:
done